app/domain/authentication/authn_azure/authenticator.rb

```diff
@@ -31,6 +32,23 @@ def validate_azure_token
       def validate_application_identity
 
       end
+
+      def provider_uri
+        azure_authenticator_secrets["provider-uri"]
+      end
+
+      def azure_authenticator_secrets
+        @azure_authenticator_secrets ||= @fetch_authenticator_secrets.(
+          service_id: service_id,
+          conjur_account: account,
+          authenticator_name: authenticator_name,
+          required_variable_names: required_variable_names
+        )
+      end
+
+      def required_variable_names
+        @required_variable_names ||= %w(provider-uri)
```

Comment:
Also why `%w` for `%w(provider-uri)` and not `%s`

To try to answer my own question, a couple of possible reasons:
1. Unity across our authenticators (this is how do it authn-oidc)
2. Set ourselves up for if we need to add another variable in the future.
3. b/c in `fetch_authenticator_secrets.rb`, we expect `required_variable_names` variable to be an array

Reply:
3 is correct. We need an array of strings. `%s` will create symbols. Read more here: https://simpleror.wordpress.com/2009/03/15/q-q-w-w-x-r-s/

Reply:
^ changed it for `%q`

Reply:
Still the same answer - we need an array of strings in `fetch_authenticator_secrets`.